pkg/controllers/follow/ingress/controller_test.go

```diff
@@ -0,0 +1,68 @@
+package ingress
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/networking/v1beta1"
+)
+
+func TestShouldNotUpdateOnEqual(t *testing.T) {
+	r := &ingressReconciler{}
+	ingress := v1beta1.Ingress{}
+	remoteIngress := v1beta1.Ingress{}
+
+	_, shouldUpdate := r.shouldUpdate(&remoteIngress, &ingress)
+	if shouldUpdate != false {
+		t.Error("Equal ingresses should not be updated")
+	}
+}
+
+func TestShouldUpdateOnSpecDiff(t *testing.T) {
+	r := &ingressReconciler{}
+	ingress := v1beta1.Ingress{
+		Spec: v1beta1.IngressSpec{
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: "test"},
+		},
+	}
+	remoteIngress := v1beta1.Ingress{
+		Spec: v1beta1.IngressSpec{
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: "test2"},
+		},
+	}
+
+	updatedIngress, shouldUpdate := r.shouldUpdate(&remoteIngress, &ingress)
+	if shouldUpdate != true {
+		t.Error("Ingresses with different spec should trigger update")
+	}
+
+	assert.Equal(t, updatedIngress.Spec, ingress.Spec)
+}
+
+func TestShouldUpdateOnAnnotationDiff(t *testing.T) {
+	r := &ingressReconciler{}
+	ingress := v1beta1.Ingress{
+		Spec: v1beta1.IngressSpec{
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: "test"},
+		},
+	}
+	ingress.Annotations = make(map[string]string)
+	ingress.Annotations = map[string]string{"annotate": "me"}
+	remoteIngress := v1beta1.Ingress{
+		Spec: v1beta1.IngressSpec{
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: "test2"},
```

Comment:
Shouldn't you use identical specs in this test?

Reply:
good catch